src/runtime/pybuffer.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Python.Runtime
+{
+    public class PyBuffer
+    {
+        private IntPtr _handle;
+        private Runtime.Py_buffer _view;
+        private GCHandle _gchandle;
+
+        internal PyBuffer(IntPtr obj, int flags, out bool success)
+        {
+            int size = Marshal.SizeOf(typeof(Runtime.Py_buffer));
+            byte[] rawData = new byte[size];
+            _gchandle = GCHandle.Alloc(rawData, GCHandleType.Pinned);
+            _handle = _gchandle.AddrOfPinnedObject();
+            _view = (Runtime.Py_buffer)Marshal.PtrToStructure(_handle, typeof(Runtime.Py_buffer));
+
+            success = Runtime.PyObject_GetBuffer(obj, _handle, flags) >= 0;
```

Comment:
Why return success/failure via `out`, and not simply throw an exception?

Reply:
... in which case you also have to release resources.

Reply:
Your right. My bad.